.github/workflows/docs.yaml

```
@@ -0,0 +1,58 @@
+name: Build Doc
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test_linux:
+    name: Linux
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+
+    - name: Configuring Test Environment
+      run: |
+        sudo apt-get update
+        sudo apt-get -y install python3.9 build-essential doxygen
+        curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+        python3.9 get-pip.py
+        python3.9 -m pip install -U pip wheel
+        python3.9 -m pip install cmake ninja
```

iirc GitHub Actions comes pre-configured with python 3.8 and pip, so you don't need to install it manually

```suggestion
        sudo apt-get -y install build-essential doxygen
        pip install ninja cmake wheel
```

---

Changed to use default Python 3.8.